src/core/managers/TransactionManager.ts

```diff
@@ -0,0 +1,149 @@
+import { Transaction } from '@multiversx/sdk-core/out';
+import axios, { AxiosError } from 'axios';
+import { BATCH_TRANSACTIONS_ID_SEPARATOR } from 'constants/transactions.constants';
+import { getAccount } from 'core/methods/account/getAccount';
+import { networkSelector } from 'store/selectors';
+import { getState } from 'store/store';
+import { GuardianActionsEnum } from 'types';
+import { BatchTransactionsResponseType } from 'types/serverTransactions.types';
+import { SignedTransactionType } from 'types/transactions.types';
+
+export class TransactionManager {
+  private static instance: TransactionManager | null = null;
+
+  private constructor() {}
+
+  public static getInstance(): TransactionManager {
+    if (!TransactionManager.instance) {
+      TransactionManager.instance = new TransactionManager();
+    }
+    return TransactionManager.instance;
+  }
+
+  public send = async (
+    signedTransactions: Transaction[] | Transaction[][]
+  ): Promise<string[] | null> => {
+    if (signedTransactions.length === 0) {
```

Comment:
maybe pass default empty array and stop returning null ?

Reply:
it will make the param optional, in stead i throw error instead of  returning null